transports/daily/src/transport.ts

```diff
@@ -41,7 +41,47 @@ export enum DailyRTVIMessageType {
   AUDIO_BUFFERING_STOPPED = "audio-buffering-stopped",
 }
 
+class DailyCallWrapper {
+  private _daily: DailyCall;
+  private _proxy: DailyCall;
+
+  constructor(daily: DailyCall) {
+    this._daily = daily;
+    this._proxy = new Proxy(this._daily, {
+      get: (target, prop, receiver) => {
+        if (typeof target[prop as keyof DailyCall] === "function") {
+          // Disable certain methods
+          if (
+            ["preAuth", "startCamera", "join", "leave", "destroy"].includes(
+              String(prop)
+            )
+          ) {
+            return () => {
+              throw new Error(`The ${String(prop)} method is disabled.`);
+            };
+          }
+          // Forward other method calls
+          return (...args: any[]) => {
+            return (target[prop as keyof DailyCall] as Function)(...args);
+          };
+        }
+        // Forward property access
+        return Reflect.get(target, prop, receiver);
+      },
+    });
+  }
+
+  get proxy(): DailyCall {
+    return this._proxy;
+  }
+
+  get instance(): DailyCall {
```

Comment:
this isn't used, right? is this just here to show how we would expose this if someday we wanted to give people "direct" (un-proxied) access to the call client?

Reply:
right. it just felt like it rounded out the implementation.

Reply:
If it were up to me I'd get rid of this since it's just dead code; I prefer to add code when it's needed rather than "just in case" we may want it someday. But don't consider this feedback blocking if you just want to merge. I'm good either way.